pkg/daemon/daemon.go

```diff
@@ -256,7 +267,22 @@ func (d *daemon) processNetworkGUID(networkID string, spec *utils.IbSriovCniSpec
 	} else {
 		guidAddr, err = d.guidPool.GenerateGUID()
 		if err != nil {
-			return fmt.Errorf("failed to generate GUID for pod ID %s, with error: %v", pi.pod.UID, err)
+			switch err {
+			// If the guid pool is exhausted, need to sync with SM in case there are unsynced changes
+			case guid.GuidPoolExhaustedError:
+				usedGuids, err := d.smClient.GetGuidsInUse()
```

Comment:
It's duplicated logic with lines #128-138. Let's move it into the separate function